sdk/core/corehttp/corehttp/runtime/policies/_authentication.py

```diff
@@ -120,18 +120,18 @@ def send(self, request: PipelineRequest[HTTPRequestType]) -> PipelineResponse[HT
         except Exception:  # pylint:disable=broad-except
             self.on_exception(request)
             raise
-        else:
-            if response.http_response.status_code == 401:
-                self._token = None  # any cached token is invalid
-                if "WWW-Authenticate" in response.http_response.headers:
-                    request_authorized = self.on_challenge(request, response)
-                    if request_authorized:
-                        try:
-                            response = self.next.send(request)
-                            self.on_response(request, response)
-                        except Exception:  # pylint:disable=broad-except
-                            self.on_exception(request)
-                            raise
+
```

Comment:
Should we also move the `self.on_response(request, response)` from the try block to here as well?

Reply:
yes, good catch